builders/txBuilder.go

```diff
@@ -32,46 +31,28 @@ func NewTxBuilder(signer Signer) (*txBuilder, error) {
 	}, nil
 }
 
-// createTransaction assembles a transaction from the provided arguments
-func (builder *txBuilder) createTransaction(arg data.ArgCreateTransaction) *data.Transaction {
-	return &data.Transaction{
-		Nonce:     arg.Nonce,
-		Value:     arg.Value,
-		RcvAddr:   arg.RcvAddr,
-		SndAddr:   arg.SndAddr,
-		GasPrice:  arg.GasPrice,
-		GasLimit:  arg.GasLimit,
-		Data:      arg.Data,
-		Signature: arg.Signature,
-		ChainID:   arg.ChainID,
-		Version:   arg.Version,
-		Options:   arg.Options,
-	}
-}
-
-// ApplySignatureAndGenerateTx will apply the corresponding sender and compute the signature field and
-// generate the transaction instance
-func (builder *txBuilder) ApplySignatureAndGenerateTx(
+// ApplySignature will apply the corresponding sender and compute and set the signature field
+func (builder *txBuilder) ApplySignature(
 	cryptoHolder core.CryptoComponentsHolder,
-	arg data.ArgCreateTransaction,
-) (*data.Transaction, error) {
-	arg.SndAddr = cryptoHolder.GetBech32()
-	unsignedMessage := builder.createUnsignedTx(arg)
+	tx *transaction.FrontendTransaction,
+) error {
+	tx.Sender = cryptoHolder.GetBech32()
+	unsignedMessage := builder.createUnsignedTx(*tx)
 
 	signature, err := builder.signer.SignTransaction(unsignedMessage, cryptoHolder.GetPrivateKey())
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	arg.Signature = hex.EncodeToString(signature)
+	tx.Signature = hex.EncodeToString(signature)
 
-	return builder.createTransaction(arg), nil
+	return nil
 }
 
 // ComputeTxHash will return the hash of the provided transaction. It assumes that the transaction is already signed,
 // otherwise it will return an error.
 // The input can be the result of the ApplySignatureAndGenerateTx function
-func (builder *txBuilder) ComputeTxHash(tx *data.Transaction) ([]byte, error) {
+func (builder *txBuilder) ComputeTxHash(tx *transaction.FrontendTransaction) ([]byte, error) {
```

Comment:
the last line of the comment is not valid anymore

Reply:
removed the last line as it is irrelevant now.